src/core/memory.d

```diff
@@ -851,12 +853,15 @@ void* pureRealloc(void* ptr, size_t size) pure @nogc nothrow
     }
     return ret;
 }
+/// ditto
+void pureFree(void* ptr) @system pure @nogc nothrow
```

Comment:
`free` is [not guaranteed not to set `errno` by current standards](https://sourceware.org/ml/libc-alpha/2015-02/msg00067.html); libc implementations may choose to set errno for both valid and invalid pointers.
POSIX.1 Issue 8 (still WIP), [is going to forbid  `free` to set `errno` for valid pointers](http://austingroupbugs.net/view.php?id=385), but explicitly allows setting it for an invalid pointers.
AFAIK there's no way for us to know whether the used libc will be linked to statically or dynamically and in the latter case there's no way AFAIK to know if the libc does or does not set `errno`.
It follows that `pureFree` must save&restore the errno value the same way that the allocation functions do:
```d
const errno = fakePureGetErrno();
fakePureFree(ptr);
if (errno != 0) fakePureSetErrno(errno);
```